nav2_costmap_2d/test/unit/denoise_layer_test.cpp

```diff
@@ -0,0 +1,647 @@
+/*********************************************************************
```

Comment:
Overall, testcase is well-made. Please check license headers and move all test-related stuff into testcase itself and it will be OK.